apps-rendering/src/client/signupForm.ts

```diff
@@ -31,13 +31,22 @@ async function handleSubmission(bundle: FormBundle): Promise<void> {
 	input.setAttribute('disabled', '');
 	submitButton.setAttribute('disabled', '');
 
-	const response = await fakeRequestToEmailSignupService(
-		input.value,
-		identityName,
-	);
+	let success: boolean | undefined = undefined;
+	// newslettersClient.requestSignUp can throw errors
+	// user feedback about the type of error (eg Network Error,
+	// Service unavailable) will be handled by the native layer
+	// The component only need to know success or failure.
+	try {
+		success = await newslettersClient.requestSignUp(
+			input.value,
+			identityName,
+		);
+	} catch (error) {
+		success = false;
+	}
 	form.classList.remove(MODIFIER_CLASSNAME.waiting);
 
-	if (response.status === 200) {
+	if (success) {
 		form.classList.add(MODIFIER_CLASSNAME.success);
 	} else {
 		form.classList.add(MODIFIER_CLASSNAME.failure);
```

Comment:
Another way we could handle the potential throw (which we've used elsewhere in AR) is to use a `.then().finally` block:

```ts
	newslettersClient.requestSignUp(input.value, identityName).then(success => {
		if (success) {
			form.classList.add(MODIFIER_CLASSNAME.success);
		} else {
			form.classList.add(MODIFIER_CLASSNAME.failure);
		}
	})
	.finally(() => {
		form.classList.remove(MODIFIER_CLASSNAME.waiting);
	})
```

that way, we could also remove the local declaration of `success`!

Reply:
thanks - that looks much better! One change I made was to include a `catch` statement to do `form.classList.add(MODIFIER_CLASSNAME.failure)` so the UI will be updated to the fail state if the function throws, as well as if it returns false.